### ISSUE TITLE
CarPlay

### ISSUE THREAD

**Issue by vCrespoP** [dennisweissmann/DeviceKit#17]
Hi man, long time no see!

I'm getting a compiling problem when using 0.3.1 (that first included CarPlay in 
```
private init() {
            switch UIDevice.currentDevice().userInterfaceIdiom {
            case .Pad:          self = .Pad
            case .Phone:        self = .Phone
            case .TV:           self = .TV
            case .Unspecified:  self = .Unspecified
            }
        }
```

But since CarPlay has been included there, the app fails when building (as it doesn't exist) and I have to stick to 0.3.0.

I wouldn't like to miss future updates of this repo because of this :D

Thanks!

**Comment by dennisweissmann** [comment]
Hey, yeah I'm reeeeeeaally sorry for that 🙊🙏, it was a mistake to include them during the beta - I hope that won't happen again ... Does that error still occur with Xcode 7.3 and a changed base SDK of 9.3?

**Comment by vCrespoP** [comment]
I was in 7.2.1, I'll come back when I update it :)

**Comment by vCrespoP** [comment]
It always take sooo long to update Xcode, I love it <3

Tested version 0.3.1 against Xcode 7.3, **working ok**.

Maybe it's good to check SDK version wherever CarPlay appears so it doesn't fail when compiling. 
But I leave this decision to the father of the framework.

**Comment by dennisweissmann** [comment]
That's *exactly* what I'm trying to do but haven't found a solution yet ... The if #available(iOS 9.3) thingy won't work ... If you know what to do please feel free!!

What I tried is

```
  @available(iOS 9.3, *)
  case CarPlay
  ...

       private init() {
          if #available(iOS 9.3, *) {
            switch UIDevice.currentDevice().userInterfaceIdiom {
            case .Pad:          self = .Pad
            case .Phone:        self = .Phone
            case .TV:           self = .TV
            case .CarPlay:      self = .CarPlay
            case .Unspecified:  self = .Unspecified
            }
          } else {
            switch UIDevice.currentDevice().userInterfaceIdiom {
            case .Pad:          self = .Pad
            case .Phone:        self = .Phone
            case .TV:           self = .TV
            case .Unspecified:  self = .Unspecified
            default:            self = .Unspecified // I think this is needed because Apple's `CarPlay` case is annotated as `@available(iOS 9.0, *)` but is only available in iOS 9.3 and later.
            }
          }
        }
```

That works fine if I link against 9.3, but not if I link against 9.2, which renders the whole `if #available` thing useless ... If you have any ideas please let me know or send a pull request!
